clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -3085,6 +3095,16 @@ ParameterStream &MemVarInfo::getKernelArg(ParameterStream &PS) {
         PS << "template ";
       PS << "get_multi_ptr<" << MapNames::getClNamespace()
          << "access::decorated::no>().get()";
+    } else if (isShared() && AccMode == PointerToArray) {
+      std::string CastType = getType()->getBaseName();
+      auto Range = getType()->getRange();
+      CastType = CastType + " (*)";
+      for (size_t i = 1; i < Range.size(); i++) {
+        CastType = CastType + "[" + Range[i].getSize() + "]";
+      }
+      PS << "reinterpret_cast<" << CastType << ">(" << getAccessorName()
```

Comment:
Use isWritten() to check, if false, we can use static_cast, otherwise, no cast needed.